src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1239,7 +1345,10 @@ def enable_model_cpu_offload(self, gpu_id: int = 0, device: Union[torch.device,
         else:
             raise ImportError("`enable_model_cpu_offload` requires `accelerate v0.17.0` or higher.")
 
-        device = torch.device(f"cuda:{gpu_id}")
+        # _offload_gpu_id should be set to passed gpu_id or default to previously set id or default to 0
+        self._offload_gpu_id = gpu_id or self._offload_gpu_id or 0
+
+        device = torch.device(f"cuda:{self._offload_gpu_id }")
```

Comment:
This might not be `cuda` if the user specified a device. Maybe use `torch.device(device).type` ?